backend/app/controllers.py

```diff
@@ -0,0 +1,234 @@
+"""
+Application controllers are found in this file.
+
+One controller per model to handle creating, updating and deleting.
+All controllers should be derived from the Base class.
+The Base contains attributes and functions common to all controllers.
+"""
+import bottle
+import common.responses as responses
+from common.utils import request_args, \
+    get_ts, rfc3339, generate_signature
+import settings
+from app import services
+from webargs import fields
+from http.cookies import Morsel
+
+
+class Base(object):
+    """Base controller class.
+
+    All controllers should derive from this class.
+    """
+    _app = None
+
+    def __init__(self, app: bottle.Bottle = None):
+        if app is not None:
+            self._app = app
+
+    def set_app(self, app: bottle.Bottle):
+        """Sets the Bottle app for the controller.
+
+        :param app: The Bottle application.
+        :type app: bottle.Bottle
+        :return: None
+        :rtype: None
+        """
+        self._app = app
+
+
+class Status(Base):
+    """Status controller.
+
+    Adds a health check endpoint to the API.
+    """
+
+    @staticmethod
+    def health_check() -> dict:
+        """Returns an empty response.
+
+        Used for API health check endpoint.
+
+        :return: An empty response.
+        :rtype: dict
+        """
+        return responses.JSONResponse({})
+
+
+class Login(Base):
+    """Auth
+
+    Handles Login API requests
+    """
+
+    def __init__(self, app: bottle.Bottle = None):
+        super().__init__(app)
+        self._request = services.Request('login')
```

Comment:
Please use names for the services instead of too generic `_request` -> `self._login = ...`